CONTRIBUTING.md

```diff
@@ -86,37 +76,46 @@ message that the code in question is special and care should be taken.
-- Make pull requests a clean representation of the implementation of a feature
-  — meandering commit history is okay for WIP branches if it's supposed to be
-  squashed down to one commit eventually, but development dead-ends and back
-  tracks while developing a feature probably aren't useful to retain. It's okay
-  if this is hard to achieve, but expect people to argue for a squash-merge.
+
+## Pull Requests
+
+- Include a newsfragment before merging. This is a file with a description of
+  your change, to go in the release notes - often less technical than a commit
+  message. This is a file numbered to match the issue or pull request in
+  the `newsfragments` folder - our current list of categories and what to call
+  this can be found [here][news-README]. These are processed on release by the
+  release manager with a tool named [towncrier]. This can be added after you've
+  created the pull request, if you don't have an issue number.
+- We aim to squash-merge most pull requests. However, if you are working on a
+  longer-term feature branch, or making a lot of changes that might be a
+  candidate for a non-squash merge - please try to keep commits a relatively
+  clean representation of the implementation of your change, by e.g. using
+  `git rebase` - This helps people to understand your changes easier.
 
 
 ## Code linting, automatic formatting and static analysis
 
-- **Please install the pre-commit hooks**. (`libtbx.precommit install` if using
-  the libtbx ecosystem). These use the [pre-commit] library and ensure that
-  various sanity checks are run before commit, including formatting, syntax
-  compatibility, basic flake8 checks, lack of conflict markers and file size
-  limits. Basically, most of the essential rules will be checked automatically
-  by this.
+- **Please use the pre-commit hooks**. If using the DIALS bootstrap installer
+  then this will be done automatically. Otherwise, use `libtbx.precommit
+  install` - if in the libtbx ecosystem - or manually install the hooks with
+  `pre-commit install`. These use the [pre-commit] tool and ensure that
+  various sanity checks are run before commit, including import order,
+  formatting, syntax compatibility, basic flake8 checks, lack of conflict
+  markers and file size limits. Basically, most of the essential rules will be
+  checked automatically by this.
 - **We format python code with [black]**. This means that while writing code
   you don't have to worry about laying things out neatly, because black will
   take care of the formatting. We prefer if you commit code formatted with
   black (the pre-commit hook will help do this for you), but if for some reason
-  you can't, the whole codebase is auto-cleaned once a week. Most IDEs and
-  editors have support for running formatters like black frequently or
+  you miss this, the whole codebase is auto-cleaned once a week. Most IDEs
+  and editors have support for running formatters like black frequently or
   automatically.
```

Comment:
```suggestion
  and editors have support for running formatters like black automatically.
```
Just shorter. And black is idempotent so it doesn't matter how frequently you run it.